src/main/java/com/kodilla/ecommercee/controller/CartController.java

```
@@ -1,71 +1,48 @@
-
-        }
-    };
 
-    @GetMapping
-    public ResponseEntity<List<CartDto>> getCart(){
-        return ResponseEntity.ok(CartDtoList);
+    @PostMapping("{userId}")
+    public void createEmptyCart(@PathVariable("userId") Long userId) {
+        cartDbService.createEmptyCart(userId);
     }
 
-    @GetMapping(value = "{cartId}")
-    public ResponseEntity<CartDto> getCart(@PathVariable Long cartId){
-        for (CartDto cart : CartDtoList) {
-            if (cart.getId().equals(cartId)){
-                return ResponseEntity.ok(cart);
-            }
-        }
-        return ResponseEntity.notFound().build();
+    @GetMapping("{userId}")
+    public List<ProductDto> getCart(@PathVariable("userId") Long userId) {
+        return productMapper.mapToProductDtoList(cartDbService.getUserCart(userId));
     }
 
-    @DeleteMapping(value = "{cartId}")
-    public ResponseEntity<Void> deleteCart(@PathVariable Long cartId) {
-        for (CartDto cart : CartDtoList) {
-            if (cart.getId().equals(cartId)){
-                CartDtoList.remove(cart);
-                return ResponseEntity.ok().build();
-            }
-        }
-        return ResponseEntity.notFound().build();
+    @PutMapping("{userId}/{productId}")
+    public void addItemToCart(@PathVariable("userId") Long userId, @PathVariable Long productId) {
+        cartDbService.addItemToCart(userId, productId);
     }
 
-    @PutMapping(path="{cartId}",consumes = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<CartDto> updateCart(@PathVariable Long cartId, @RequestBody CartDto cartDto) {
-        for (CartDto cart : CartDtoList) {
-            if (cart.getId().equals(cartId)) {
-                CartDtoList.remove(cart);
-                CartDtoList.add(cartDto);
-                return ResponseEntity.ok(cartDto);
-            }
-        }
-        return ResponseEntity.notFound().build();
+    @DeleteMapping("{userId}/{productId}")
```

The same - I think we should throw an exception here.

---

Added